docs/packages/context-menu/Context-menu-manager.md

```diff
@@ -0,0 +1,345 @@
+		badge: null,		// Optional, if not specified will be null as default.
+		action: null,		// Optional, if not specified will be null as default.
+		onMouseEnter: null,	// Optional, if not specified will be null as default.
+		onMouseLeave: null,	// Optional, if not specified will be null as default.
+ 	}
+ );
+```
+
+&nbsp;  
+Notes:
+> The groupId parameter must point to an existing group.
+
+> The new item will reside inside its group parent 'items[]'. 
+
+> The item 'id' property must be unique. recommended using as: nameOfGroup_itemAction.
+
+>  The item 'label' property is important, it describes the items action.
+ 
+> The item 'label' property can contain up to 30 chars max (or will be trimmed).
+
+
+&nbsp;
+### Update item
+`updateItem( groupId, itemId, itemData )` Method updates an item data.
+
+&nbsp;  
+Example:
+```JS
+updateItem(
+	'groupId',					// Required.
+	'itemId',					// Required.
+ 	{
+		label: 'action name',	// One or more item properties to be update.
+ 	}
+ );
+```
+
+&nbsp;  
+Notes:
+> The groupId parameter must point to an existing group.
+
+> The itemId parameter must point to an existing item inside the given group.
+
+
+&nbsp;
+### Delete item
+`deleteItem( groupId, itemId )` Method deletes an item.
+
+&nbsp;  
+Example:
+```JS
+deleteItem(
+	'groupId',					// Required.
+	'itemId',					// Required.
+ );
+```
+
+&nbsp;  
+Notes:
+> The item will be deleted with all the subgroups that points to it.
+
+
+&nbsp;
+### Get config
+`getConfig()` Method returns the config tree of groups & items.
+This method is used to get the config tree for building the context menu structure in the UI.
+It returns an array of groups, that contains items, that may contain subgroups and so on.
+
+
+&nbsp;
+## Understanding the usage flow
+
+The config tree is built as an array because of the need to manage order of groups & items in the tree.
+The manager is built to allow prioritizing built-in element registering new groups over 3rd party registrations.
+To better understand that, lets dive into the registration of new groups & the triggering flow works:
+
+### Registration of new items
+
+There are two cases for adding items to the context menu. the first case is for built-in widgets & elements in the editor (column, section, containers, widgets) that add items to the context menu, and the second is when a third-party extension/plugin does so. Both of them use the same API methods of the contextMenuManager.
+
+The built-in element has the priority to register its items first, and also to register the last items too. In fact, the built-in element manages the timing of third-parties registration and triggers the entire flow.
+
+### Triggering the context menu flow
+
+The element triggers the `GlobalEventHandlers.oncontextmenu` when a right-click or context-menu keyboard key down happens inside the element scope. the `oncontextmenu` calls a function that starts the flow as described:
```

Comment:
```suggestion
The element triggers a [`contextmenu`](https://developer.mozilla.org/en-US/docs/Web/API/Element/contextmenu_event) event on right-click or context-menu key click inside its scope. The `oncontextmenu` calls a function that starts the flow as described:
```

Plus I didn't really understand the last sentence

Reply:
Partially accepted. The 'context-menu keyboard key down' is a specific keyboard key on the keyboard...